langchain4j-redis/src/main/java/dev/langchain4j/store/embedding/redis/RedisChatMemoryStore.java

```diff
@@ -0,0 +1,62 @@
+package dev.langchain4j.store.embedding.redis;
+
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.data.message.ChatMessageDeserializer;
+import dev.langchain4j.data.message.ChatMessageSerializer;
+import dev.langchain4j.store.memory.chat.ChatMemoryStore;
+import lombok.Builder;
+import redis.clients.jedis.JedisPooled;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static dev.langchain4j.internal.ValidationUtils.*;
+
+public class RedisChatMemoryStore implements ChatMemoryStore {
+
+    private final JedisPooled client;
+
+    @Builder
+    public RedisChatMemoryStore(String host,
+                                Integer port,
+                                String user,
+                                String password) {
+        String finalHost = ensureNotBlank(host, "host");
+        int finalPort = ensureNotNull(port, "port");
+        if (user != null) {
```

Comment:
we could evaluate null or empty, but probably not a big deal.

Reply:
It's ok to be null, the client will be created without user/pass. But if the user is empty I think it should throw an exception because the intention was to have an user, that's why I evaluated only in the next line.